lib/sprockets/archiving.rb

```diff
@@ -0,0 +1,72 @@
+require 'sprockets/http_utils'
+require 'sprockets/processor_utils'
+require 'sprockets/utils'
+
+module Sprockets
+  module Archiving
+    include Utils
+
+    def archivers
+      config[:archivers]
+    end
+
+    # Public: Register archiver, it has to respond to 3 methods
+    # type can be any symbol
+    #
+    # register_archiver :gzip, Utils::Zlib 
+    #
+    # will register Zlib compression for text file, 
+    # we dont use myme_types here, because different mime_type 
+    # can be compressed by one archiver (text/stylesheets, text/javascript, etc)
+    def register_archiver type, archiver
+      if archiver.is_a?( Class )
+        klass = archiver
+      else
+        raise(Error, "archiver is'n a class: #{archiver}")
+      end
+
+      self.config = hash_reassoc(config, :archivers) do |archivers|
+        archivers[type] = klass
+        archivers
+      end
+    end
+
+    def unregister_archiver type
+      self.config = hash_reassoc(config, :archivers) do |archivers|
+        archivers.delete(type)
+        archivers
+      end
+    end
+
+    # Public: Checks if Gzip is enabled.
+    def gzip?
```

Comment:
Since this is not coupled to gzip anymore we may need to change these methods. Like `archiver_enabled?` `skip_archiver?` and `archiver=`.